sled-agent/src/bootstrap/agent.rs

```diff
@@ -263,11 +263,20 @@ impl Agent {
 
             return Ok(SledAgentResponse { id: server.id() });
         }
+
+        // TODO(https://github.com/oxidecomputer/omicron/issues/823):
+        // Currently, the prescence or abscence of RSS is our signal
+        // for "is this a scrimlet or not".
+        // Longer-term, we should make this call based on the underlying
```

Comment:
Just a note that this changes at runtime with respect to hardware. It's not a static thing and the underlying hardware presence will come and go as the Sidecar is power cycled.

Reply:
That matches my understanding. I do think there is more work that'll need to smooth out the case of "a sled has been removed from one location in the rack, and plugged in elsewhere" - perhaps a call to Nexus, to figure out what services should change - but otherwise, this makes sense.

Reply:
Yeah, while the movement bit seemed captured, I just wanted to make sure it was clear that you might choose to start a bootstrap agent before this is known and wanted to call that out as part of this given what appeared the initial start the agent piece was encoding this knowledge right now.

Reply:
Makes sense, thanks Robert. I believe Sean added functionality to update the database's record including the `is_scrimlet` value in [d172c9c](https://github.com/oxidecomputer/omicron/pull/1510/commits/d172c9c4c6f11f24a33b99a75e582762372cbd4e).

One question I have though is about when that value is updated. At RSS-time, it's definitely possible that the sled agent comes up before Sidecar, so we'd need to support updating it from `false` to `true`. But it seems kind of "sticky" to me. That is, I'm not sure it makes sense to set this to `false`, at least not right away, when Sidecar is cycled. If that's done as part of an administrative action or by Nexus, it seems like we'd want to keep the knowledge that this is or was a Scrimlet, and report 503s from `dpd` or the sled agent. If Sidecar cycles unexpectedly, we probably still want to do that, along with generating an alert of some kind. I don't know if there's any action here, but spelling that out seems useful to me.

Reply:
I suspect an initial policy of stickiness across a given boot will be a pretty useful way to go here. That is, when it's rebooted, it'll be something that is determined fresh and resets stickiness.